aws-android-sdk-pinpoint/src/main/java/com/amazonaws/mobileconnectors/pinpoint/targeting/notification/EventSourceType.java

```diff
@@ -0,0 +1,204 @@
+/**
+ * Copyright 2016-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazonaws.mobileconnectors.pinpoint.targeting.notification;
+
+import android.os.Bundle;
+import android.util.Log;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonPrimitive;
+import com.google.gson.JsonSyntaxException;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.amazonaws.mobileconnectors.pinpoint.targeting.notification.NotificationClientBase.CAMPAIGN_ACTIVITY_ID_ATTRIBUTE_KEY;
+import static com.amazonaws.mobileconnectors.pinpoint.targeting.notification.NotificationClientBase.CAMPAIGN_ID_ATTRIBUTE_KEY;
+import static com.amazonaws.mobileconnectors.pinpoint.targeting.notification.NotificationClientBase.CAMPAIGN_PUSH_KEY_PREFIX;
+import static com.amazonaws.mobileconnectors.pinpoint.targeting.notification.NotificationClientBase.JOURNEY_ACTIVITY_ID_ATTRIBUTE_KEY;
+import static com.amazonaws.mobileconnectors.pinpoint.targeting.notification.NotificationClientBase.JOURNEY_ID_ATTRIBUTE_KEY;
+
+class EventSourceType {
+    private static final String TAG = EventSourceType.class.getSimpleName();
+    private static final String AWS_EVENT_TYPE_OPENED = "opened_notification";
+    private static final String AWS_EVENT_TYPE_RECEIVED_FOREGROUND = "received_foreground";
+    private static final String AWS_EVENT_TYPE_RECEIVED_BACKGROUND = "received_background";
+    private static final String PINPOINT_ATTRIBUTE_KEY = "pinpoint";
+    private static final String JOURNEY_ATTRIBUTE_KEY = "journey";
+    private static final String CAMPAIGN_EVENT_SOURCE_NAME = "campaign";
+    private static final String JOURNEY_EVENT_SOURCE_NAME = "journey";
+    private static final String CAMPAIGN_EVENT_SOURCE_PREFIX = "_campaign";
+    private static final String JOURNEY_EVENT_SOURCE_PREFIX = "_journey";
+    static final String UNKNOWN_EVENT_SOURCE_NAME = "unknown";
```

Comment:
Not `private`?

Reply:
I was referencing this field in one single place to determine if the event type is unknown. I went ahead and added a method called isUnkown that does the comparison and was able to mark this field as private. A little cleaner that way